p2p/addrbook_provider/persistence.go

```diff
@@ -69,9 +71,13 @@ func (pabp *persistenceAddrBookProvider) GetStakedAddrBookAtHeight(height uint64
 	return addrBook, nil
 }
 
-func (pabp *persistenceAddrBookProvider) ActorsToAddrBook(actors map[string]modules.Actor) (typesP2P.AddrBook, error) {
+func (pabp *persistenceAddrBookProvider) ActorsToAddrBook(actors map[string]coreTypes.Actor) (typesP2P.AddrBook, error) {
 	book := make(typesP2P.AddrBook, 0)
 	for _, v := range actors {
+		// only add validator actors since they are the only ones having a service url in their generic param at the moment
+		if v.ActorType != coreTypes.ActorType_ACTOR_TYPE_VAL {
```

Comment:
Related to one of my other comments, fisherman and serviceNodes also have serviceURLs in their parameter, and can (and should) also participate in structured broadcast.

Should they not be included as well?

![Screenshot 2023-01-03 at 11 49 05 AM](https://user-images.githubusercontent.com/1892194/210402625-0a00fcb1-2c66-42a9-b877-f2d1fff247ce.png)

Reply:
TL;DR: The way I see it, **NO**... 

Like you [said](https://github.com/pokt-network/pocket/pull/403#discussion_r1060764619), it looks like these `serviceUrl`s have a different meaning depending on the actor type and therefore using them to connect to a node for consensus related messaging/structured broadcast is a... shortcut.

Nodes are listening on the `ConsensuPort` which by default is 8080 and it seems that `serviceUrl` has been used to have an easy way to address them.

I guess this is because we don't have -yet- a full-fledged P2P module that owns the addressbook and can be relied upon for routing and messaging (that will come with peer discovery and churn).

We won't address messages via `serviceUrl` but rather by `address` or `nodeId`, then P2P would figure out how to reach that peer.

In `LocalNet` we spin up 4 validator nodes and the debug commands are all related to consensus.
This explains why previously we had 👇👈 while now we have 👉👇
![image](https://user-images.githubusercontent.com/29378614/210437954-08ad4cba-acec-49a5-9801-89ba9765ae05.png)

If we consider other actors I can only imagine that since we have overlaps (eg: `node1.consensus:8080` in `genesis.json` is a Validator, a ServiceNode and also a Fisherman) we would have duplicate messages at a minimum.

I wish Brian was here to pick his brain on this :) the ways I see it, structured gossip happens under the hood in the P2P module, when you say "broadcast to all validators" you don't know/care if the message is also handled and retransmitted by a Fisherman in order to reach a "nearby" Validator. That's implementation detail of the P2P algorithm.

I would say that for now we have to filter. Happy to discuss further even offline. 

Just let me know how to proceed here.

Reply:
Solid explanation. I'm convinced.

Excited to see what our next steps are going to be here :)

Reply:
Added https://github.com/pokt-network/pocket/issues/426 PTAL 🙏